Spigot-Server-Patches/0103-Add-handshake-event-to-allow-plugins-to-handle-clien.patch

```diff
@@ -28,15 +28,15 @@ index 8f2289f4e..93ca93b64 100644
 +                            return;
 +                        }
 +
-+                        packethandshakinginsetprotocol.hostname = event.getServerHostname();
-+                        this.b.socketAddress = new java.net.InetSocketAddress(event.getSocketAddressHostname(), ((java.net.InetSocketAddress) this.b.getSocketAddress()).getPort());
++                        if (event.getServerHostname() != null) packethandshakinginsetprotocol.hostname = event.getServerHostname();
++                        if (event.getSocketAddressHostname() != null) this.b.socketAddress = new java.net.InetSocketAddress(event.getSocketAddressHostname(), ((java.net.InetSocketAddress) this.b.getSocketAddress()).getPort());
 +                        this.b.spoofedUUID = event.getUniqueId();
 +                        this.b.spoofedProfile = gson.fromJson(event.getPropertiesJson(), com.mojang.authlib.properties.Property[].class);
 +                        handledByEvent = true; // Hooray, we did it!
 +                    }
 +                }
 +                // Don't try and handle default logic if it's been handled by the event.
-+                if (!handledByEvent && proxyLogicEnabled) {
++                if (!handledByEvent && org.spigotmc.SpigotConfig.bungee) {
```

Comment:
This is not a safe change, if you want to expose something to the event, it should be handled via the event